src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Android.cs

```diff
@@ -89,6 +90,12 @@ private static unsafe bool VerifyRemoteCertificate(IntPtr sslStreamProxyHandle)
                 try
                 {
                     proxy.ValidationResult = proxy._sslStream.VerifyRemoteCertificate();
+                    if (!proxy.ValidationResult.IsValid)
+                    {
+                        // Throw AuthenticationException if validation failed
+                        proxy.ValidationException = new AuthenticationException(SR.net_ssl_io_cert_custom_validation);
+                        return false;
+                    }
                     return proxy.ValidationResult.IsValid;
```

Comment:
This will always be true... maybe we should remove the return from the if branch and let it fall through to the return here.